src/server/services/saml.strategy.ts

```diff
@@ -0,0 +1,47 @@
+import { Strategy } from 'passport-saml';
+import { PassportStrategy } from '@nestjs/passport';
+import { Injectable, UnauthorizedException } from '@nestjs/common';
+import { UserData, HarvardKeyProfile } from '../interfaces';
+import { ConfigService } from './config.service';
+
+/**
+ * Implements passport-saml to connect to Harvard Key for authentication
+ */
+
+@Injectable()
+class SAMLStrategy extends PassportStrategy(Strategy) {
+  private readonly devMode: boolean;
+
+  public constructor(config: ConfigService) {
+    super({
+      entryPoint: config.get('CAS_URL'),
+      issuer: 'passport-saml',
+      host: config.get('EXTERNAL_URL'),
+    });
+    this.devMode = !config.isProduction;
+  }
+
+  public async validate(profile: HarvardKeyProfile): Promise<UserData> {
+    if (this.devMode) {
```

Comment:
I wonder about extracting this out into a separate instance that always returns true that we can use in dev mode.

Reply:
Opted not to do this since it would just move the `if this.devMode` to a custom provider without providing any real visible benefit